moveit_core/constraint_samplers/test/pr2_arm_ik.cpp

```diff
@@ -785,5 +785,5 @@ bool PR2ArmIK::checkJointLimits(const double& joint_value, const int& joint_num)
   else
     jv = angles::normalize_angle(joint_value * angle_multipliers_[joint_num]);
 
-  return not(jv < min_angles_[joint_num] || jv > max_angles_[joint_num]);
+  return !(jv < min_angles_[joint_num] || jv > max_angles_[joint_num]);
```

Comment:
oh the classic inability of MS to properly implement C++ ...

Reply:
I'm not a fan of mixing the operator types anyway (why `not`, but not `or`), so we might as well fix it.

Reply:
sure, I just think C++ would look half as ugly if `not`, `and`, `or` could be used in instead of "random" special characters ... but since "Basic for MS Dos" back in 1843 screwed that up now we cannot ever use that ...

Reply:
It seems you just have to disable msvc specific extensions which sounds like a good idea generally: https://docs.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-160&viewFallbackFrom=vs-2019

Reply:
I would have expected [this cmake option](https://github.com/ros-planning/moveit/blob/master/moveit_core/CMakeLists.txt#L8), but apparently msvc's non-permissive mode is something different than standard extensions...

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/17068 talks about that but is still open, it also says this is the default now in VS2017 projects

Reply:
I think you should debate that with the catkin folks or whoever sets global variables like that.

Since this is the only "not" in the codebase (and also the only one in all of ROS as far as I can see) I think the best course of action is to change it to !

Reply:
Sure, no real objections for this specific case here, my remarks were more of a general nature.